File: PDOO/p4/civitas_rb/lib/diario.rb
require 'singleton'
 
module Civitas
class Diario
  include Singleton
  
  def initialize
    @eventos=[]
  end
  
  Diario.instance
  
  def diario
    @eventos=Array.new
  end
  
  def ocurre_evento(e)
    @eventos.push(e)
  end
  
  def eventos_pendientes
    return !@eventos.empty?
  end
  
  def leer_evento
    salida=""
    if(!@eventos.empty?)
      salida= @eventos.delete_at(0)
    end
    return salida
  end
  
  attr_accessor :eventos
  
  private :diario
end
end

File: PDOO/PDOO/practica emilio/Civitas/src/civitas/Diario.java
package civitas;

import java.util.ArrayList;

public class Diario {
  //Atributos de clase
  static final private Diario instance = new Diario();
  //Atributos de instancia
  private final ArrayList<String> eventos;
  
  static public Diario getInstance() {
    return instance;
  }

    public ArrayList<String> getEventos() {
        return eventos;
    }
  
  
  
  public Diario () {
    eventos = new ArrayList<>();
  }
  
  void ocurreEvento (String e) {
    eventos.add (e);
  }
  
  public boolean eventosPendientes () {
    return !eventos.isEmpty();
  }
  
  public String leerEvento () {
    String salida = "";
    if (!eventos.isEmpty()) {
      salida = eventos.remove(0);
    }
    return salida;
  }
}
